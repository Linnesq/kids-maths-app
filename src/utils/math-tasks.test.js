const { 
    getSimpleAddition,
    getSimpleMultiplication,
    getSimpleDivision,
    getSimpleSubtraction
 } = require('./math-tasks.js');
let mockMath;

describe('MathTasks module', () => {

    beforeEach(() => mockMath = jest
        .spyOn(global.Math, 'random')
        .mockReturnValue(0.2));

    afterEach(() => mockMath.mockRestore());

    it('getSimpleAddition should return an expected object', () => {
        expect(getSimpleAddition()).toMatchSnapshot();
    });

    it('userFills random, getSimpleAddition should return an expected object', () => {
        expect(getSimpleAddition('random')).toMatchSnapshot();
    });

    it('getSimpleMultiplication should return an expected object', () => {
        expect(getSimpleMultiplication()).toMatchSnapshot();
    });

    it('getSimpleDivision should return an expected object', () => {
        expect(getSimpleDivision()).toMatchSnapshot();
    });

    it('getSimpleSubtraction should return an expected object', () => {
        expect(getSimpleSubtraction()).toMatchSnapshot();
    });
});