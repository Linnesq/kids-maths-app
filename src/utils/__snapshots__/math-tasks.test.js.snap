// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MathTasks module getSimpleAddition should return an expected object 1`] = `
Object {
  "answer": 6,
  "left": 3,
  "operator": "+",
  "right": 3,
  "userFills": "right",
}
`;

exports[`MathTasks module getSimpleDivision should return an expected object 1`] = `
Object {
  "answer": 3,
  "left": 9,
  "operator": "÷",
  "right": 3,
  "userFills": "right",
}
`;

exports[`MathTasks module getSimpleMultiplication should return an expected object 1`] = `
Object {
  "answer": 9,
  "left": 3,
  "operator": "×",
  "right": 3,
  "userFills": "right",
}
`;

exports[`MathTasks module getSimpleSubtraction should return an expected object 1`] = `
Object {
  "answer": 5,
  "left": 5,
  "operator": "−",
  "right": 0,
  "userFills": "right",
}
`;

exports[`MathTasks module userFills random, getSimpleAddition should return an expected object 1`] = `
Object {
  "answer": 6,
  "left": 3,
  "operator": "+",
  "right": 3,
  "userFills": "left",
}
`;
